### PR TITLE
emacs: append /run/wrappers/bin to tramp-remote-path

### PR DESCRIPTION
**Pull request by vikanezrimaya**
## Description of changes

This fixes cd-ing to `/sudo:localhost:` in eshell and executing commands in that eshell via TRAMP.

Test case:
1. `M-x eshell`
2. `cd /sudo:localhost:`
3. Run any command that is not an eshell builtin

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vikanezrimaya** [comment]
Oops, just realized my elisp is completely wrong.

File: pkgs/applications/editors/emacs/site-start.el [hunk 1]
```diff
@@ -68,7 +68,10 @@ least specific (the system profile)"
   ;; TODO: We should also add the other `NIX_PROFILES' to this path.
   ;; However, these are user-specific, so we would need to discover
   ;; them dynamically after connecting via `tramp'
-  '(add-to-list 'tramp-remote-path "/run/current-system/sw/bin"))
+  '(progn
+     (add-to-list 'tramp-remote-path "/run/current-system/sw/bin")
+     (add-to-list 'tramp-remote-path "/run/wrappers/bin"))
+  )
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/site-start.el:74]: In lisp, `)` is almost never put in a new line.
```suggestion
     (add-to-list 'tramp-remote-path "/run/wrappers/bin")))
```

**Comment by jian-lin** [comment]
BTW， `ping` is not a good example on modern systems (with systemd-243 or newer) where `ping` does not any privilege.  A related PR https://github.com/NixOS/nixpkgs/pull/253764.

**Comment by vikanezrimaya** [comment]
@jian-lin done. Reworded the commit message a just a little to remove the `ping` reference.

**Review by jian-lin — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**